source/updateCheck.py

```diff
@@ -359,26 +402,59 @@ def onLaterButton(self, evt):
 		saveState()
 		self.Close()
 
-class UpdateAskInstallDialog(wx.Dialog):
+	def onReviewAddonsButton(self, evt):
+		from gui import addonGui
+		incompatibleAddons = addonGui.IncompatibleAddonsDialog(
+			parent=self,
+			NVDAVersion=self.newNVDAVersionTuple
+		)
+		incompatibleAddons.ShowModal()
+		incompatibleAddons.Destroy()
+
+class UpdateAskInstallDialog(wx.Dialog, DpiScalingHelperMixin):
 
 	def __init__(self, parent, destPath, version):
 		self.destPath=destPath
 		self.version = version
-		storeUpdatesDirWritable=os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir, os.W_OK)
+		self.storeUpdatesDirWritable=os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir, os.W_OK)
 		# Translators: The title of the dialog asking the user to Install an NVDA update.
-		super(UpdateAskInstallDialog, self).__init__(parent, title=_("NVDA Update"))
+		wx.Dialog.__init__(self, parent, title=_("NVDA Update"))
+		DpiScalingHelperMixin.__init__(self, self.GetHandle())
 		mainSizer = wx.BoxSizer(wx.VERTICAL)
 		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
-		
 		# Translators: A message indicating that an updated version of NVDA is ready to be installed.
-		sHelper.addItem(wx.StaticText(self, label=_("NVDA version {version} is ready to be installed.\n").format(version=version)))
+		message = _("NVDA version {version} is ready to be installed.\n").format(version=version)
+
+		newNVDAVersionTuple = versionInfo.getNVDAVersionTupleFromString(self.version)
+		addonsWithoutKnownCompat = list(getAddonsWithoutKnownCompatibility(newNVDAVersionTuple))
+		showAddonCompat = any(addonsWithoutKnownCompat)
+		if showAddonCompat:
+			# Translators: A message indicating that some add-ons will be disabled unless reviewed before installation.
+			message = message + _(
+				"\nHowever, your NVDA configuration contains add-ons that are not tested with this version of NVDA. "
+				"These add-ons will be disabled after installation. "
+				"If you rely on these add-ons, please review the list to manually enable them before installation."
+			)
+			for a in addonsWithoutKnownCompat:
+				# now that the use is warned about the compatibility and so that the user is
+				# not prompted again after installation, we set the default compatibility
+				AddonCompatibilityState.setAddonCompatibility(
+					addon=a,
+					NVDAVersion=newNVDAVersionTuple,
+					compatibilityStateValue=compatValues.MANUALLY_SET_INCOMPATIBLE)
+		text = sHelper.addItem(wx.StaticText(self, label=message))
+		text.Wrap(self.scaleSize(500))
 
 		bHelper = sHelper.addDialogDismissButtons(guiHelper.ButtonHelper(wx.HORIZONTAL))
 		# Translators: The label of a button to install an NVDA update.
 		installButton = bHelper.addButton(self, wx.ID_OK, label=_("&Install update"))
 		installButton.Bind(wx.EVT_BUTTON, self.onInstallButton)
 		installButton.SetFocus()
-		if storeUpdatesDirWritable:
+		if showAddonCompat:

```

Comment:
Same concern as earlier, one might miss the button, so it might help to have this before the install button.